std/algorithm/mutation.d

```diff
@@ -562,6 +563,13 @@ void fill(Range, Value)(Range range, Value value)
     }
 }
 
+/// ditto
+void fill(A, Value)(auto ref A array, auto ref Value value)
+if (is(typeof((){array[] = value;})) && isSomeChar!Value && isMutable!Value)
```

Comment:
- By merging with the upper overload we loose the ability to fill static arrays.
- isMutable can well be stripped from the constraint.